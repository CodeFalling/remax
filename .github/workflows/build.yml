name: Node CI

on: [push]

jobs:
  build:

    runs-on: ${{ matrix.os }}

    strategy:
      matrix:
        node-version: [10.x, 12.x]
        os: [ubuntu-latest, windows-latest]
      fail-fast: false

    steps:
    - uses: actions/checkout@v1
    - name: Use Node.js ${{ matrix.node-version }}
      uses: actions/setup-node@v1
      with:
        node-version: ${{ matrix.node-version }}
    - name: install
      run: yarn install
    - name: lint
      run: yarn lint --fix
    - name: build
      run: yarn build
    - name: test
      run: yarn test:cov
      env:
        CI: true
    - name: Upload coverage to Codecov
      if: success() && runner.os == 'Linux'
      run: curl -s https://codecov.io/bash | bash -s -- -t $CODECOV_TOKEN
      env:
        CODECOV_TOKEN: ${{secrets.CODECOV_TOKEN}}
