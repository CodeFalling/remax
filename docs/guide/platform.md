---
title: 平台
order: 28
---

Remax 目前支持的平台

- 微信小程序: wechat
- 支付宝小程序: alipay

## 获取平台信息

```js
import { Platform } from 'remax';

console.log(Platform.current) // 当前环境，如 wechat，alipay 或其他
console.log(Platform.isWechat) // 是否是微信
console.log(Platform.isAlipay) // 是否是支付宝

...
```

## 针对平台开发

Remax 为每个小程序平台提供单独的模块，如：

```js
// 微信小程序开发
import { View, navigateTo, useShow } from 'remax/wechat';
```

```js
// 支付宝小程序开发
import { View, navigateTo, useShow } from 'remax/alipay';
```

## 跨平台开发

如果你有跨平台开发的需求，请查阅 [高级指南 - 跨平台开发](/advanced-guide/cross-platform)
