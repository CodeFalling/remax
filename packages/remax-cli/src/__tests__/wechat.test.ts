import runTest from './helpers/runTest';
import { Platform } from '../build/platform';

describe('build wechat app', () => {
  runTest('wechat', Platform.wechat);
});
