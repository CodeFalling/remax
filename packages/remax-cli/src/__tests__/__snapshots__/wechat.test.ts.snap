// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wechat build simple app 1`] = `
Array [
  Object {
    "code": "'use strict';

App({});
",
    "fileName": "__tests__/fixtures/simple/src/app.js",
  },
  Object {
    "code": "'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var index$2 = require('../../../../../../npm/react/index.js');
require('../../../../../../npm/remax/lib/render.js');
require('../../../../../../npm/remax/lib/createAppConfig.js');
var createPageConfig = require('../../../../../../npm/remax/lib/createPageConfig.js');
require('../../../../../../npm/remax/lib/index.js');
var View = require('../../../../../../npm/remax/lib/adapters/wechat/components/View.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Input.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Textarea.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Video.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Swiper.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/ScrollView.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/SwiperItem.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/MovableView.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/MovableArea.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/CoverView.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/CoverImage.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Icon.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Text.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/RichText.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Progress.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Button.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/CheckboxGroup.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Checkbox.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Form.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Label.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Picker.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/PickerView.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/PickerViewColumn.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/RadioGroup.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Radio.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Slider.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Switch.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Navigator.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Image.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/LivePlayer.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/LivePusher.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Map.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Canvas.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/OpenData.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/OfficialAccount.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Editor.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Audio.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/Ad.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/WebView.js');
require('../../../../../../npm/remax/lib/adapters/wechat/components/FunctionalPageNavigator.js');
require('../../../../../../npm/remax/lib/adapters/wechat/api.js');

var _page = function _page() {
  return index$2.createElement(View.default, null, \\"foo\\");
};

var index = Page(createPageConfig.default(_page));

exports.default = index;
",
    "fileName": "__tests__/fixtures/simple/src/pages/wechat/index.js",
  },
  Object {
    "code": "<template name=\\"REMAX_TPL\\">
  <block wx:for=\\"{{tree.root.children}}\\" wx:key=\\"{{id}}\\">
    <template is=\\"REMAX_TPL_1_CONTAINER\\" data=\\"{{i: item}}\\" />
  </block>
</template>

<wxs module=\\"_h\\">
  module.exports = {
  v: function(value) {
  return value !== undefined ? value : '';
  }
  };
</wxs>






<template name=\\"REMAX_TPL_1_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_2_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_1_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_1_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_1_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_2_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_3_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_2_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_2_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_2_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_3_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_4_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_3_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_3_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_3_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_4_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_5_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_4_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_4_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_4_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_5_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_6_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_5_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_5_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_5_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_6_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_7_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_6_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_6_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_6_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_7_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_8_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_7_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_7_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_7_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_8_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_9_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_8_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_8_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_8_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_9_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_10_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_9_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_9_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_9_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_10_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_11_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_10_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_10_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_10_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_11_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_12_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_11_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_11_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_11_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_12_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_13_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_12_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_12_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_12_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_13_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_14_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_13_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_13_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_13_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_14_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_15_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_14_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_14_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_14_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_15_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_16_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_15_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_15_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_15_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_16_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_17_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_16_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_16_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_16_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_17_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_18_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_17_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_17_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_17_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_18_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_19_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_18_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_18_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_18_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_19_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_20_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_19_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_19_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_19_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>





<template name=\\"REMAX_TPL_20_view\\">
  <view >
    
    <block wx:for=\\"{{i.children}}\\" wx:key=\\"{{id}}\\">
      <template is=\\"REMAX_TPL_21_CONTAINER\\" data=\\"{{i: item}}\\" />
    </block>
    
  </view>
</template>

<template name=\\"REMAX_TPL_20_plain-text\\" data=\\"{{i: i}}\\">
  <block>{{i.text}}</block>
</template>
<!--  把动态选择模板的逻辑放入一个模板内，可以提升性能问题 -->
<template name=\\"REMAX_TPL_20_CONTAINER\\" data=\\"{{i: i}}\\">
  <template is=\\"{{'REMAX_TPL_20_' + i.type}}\\" data=\\"{{i: i}}\\" />
</template>
",
    "fileName": "base.wxml",
  },
  Object {
    "code": "{
  \\"pages\\": [
    \\"pages/wechat/index\\"
  ],
  \\"window\\": {
    \\"navigationBarTitleText\\": \\"Wechat Todo App\\",
    \\"navigationBarBackgroundColor\\": \\"#323239\\"
  }
}",
    "fileName": "app.json",
  },
  Object {
    "code": "var tree = {
  root: {
    children: [],
  },
};

function reduce(action) {
  switch (action.type) {
    case 'splice':
      for (var i = 0; i < action.payload.length; i += 1) {
        var value = get(tree, action.payload[i].path);
        if (action.payload[i].item) {
          value.splice(
            action.payload[i].start,
            action.payload[i].deleteCount,
            action.payload[i].item
          );
        } else {
          value.splice(action.payload[i].start, action.payload[i].deleteCount);
        }
        set(tree, action.payload[i].path, value);
      }
      return tree;
    default:
      return tree;
  }
}

function getKey(key) {
  var intKey = parseInt(key);
  if (intKey.toString() === key) {
    return intKey;
  }
  return key;
}

function set(obj, path, value) {
  if (typeof path === 'string') {
    path = path.split('.').map(getKey);
  }

  if (path.length === 1) {
    obj[path[0]] = value;
  }

  var nextObj = obj;
  for (var i = 0; i < path.length; i += 1) {
    var currentPath = path[i];
    var currentValue = nextObj[currentPath];

    if (currentValue === void 0) {
      //check if we assume an array
      if (typeof path[i + 1] === 'number') {
        nextObj[currentPath] = [];
      } else {
        nextObj[currentPath] = {};
      }
    }

    if (i === path.length - 1) {
      nextObj[currentPath] = value;
    }

    nextObj = nextObj[currentPath];
  }
}

function get(obj, path) {
  if (typeof path === 'string') {
    path = path.split('.').map(getKey);
  }

  var nextObj = obj;
  for (var i = 0; i < path.length; i += 1) {
    var currentPath = path[i];
    nextObj = nextObj[currentPath];
    if (nextObj === void 0) {
      if (currentPath === 'children') {
        nextObj = [];
      } else {
        nextObj = {};
      }
    }
  }

  return nextObj;
}


module.exports = {
  reduce: reduce,
};

",
    "fileName": "helper.wxs",
  },
  Object {
    "code": "<wxs src=\\"../../../../../../helper.wxs\\" module=\\"helper\\" />
<import src=\\"../../../../../../base.wxml\\"/>
<template is=\\"REMAX_TPL\\" data=\\"{{tree: helper.reduce(action)}}\\" />
",
    "fileName": "__tests__/fixtures/simple/src/pages/wechat/index.wxml",
  },
]
`;
